Xamarin.Essentials/SaveToGallery/SaveToGallery.android.cs

```diff
@@ -0,0 +1,113 @@
+            values.Put(MediaStore.MediaColumns.DisplayName, newFileName);
+
+            var mimeType = MimeTypeMap.Singleton.GetMimeTypeFromExtension(extension.Replace(".", string.Empty));
+            if (!string.IsNullOrWhiteSpace(mimeType))
+                values.Put(MediaStore.MediaColumns.MimeType, mimeType);
+
+            using var externalContentUri = type == MediaFileType.Image
+                ? MediaStore.Images.Media.ExternalContentUri
+                : MediaStore.Video.Media.ExternalContentUri;
+
+            var relativePath = type == MediaFileType.Image
+                ? Environment.DirectoryPictures
+                : Environment.DirectoryMovies;
+
+            if (Platform.HasApiLevelQ)
+            {
+#if MONOANDROID10_0
+                values.Put(MediaStore.MediaColumns.RelativePath, Path.Combine(relativePath, albumName));
+                values.Put(MediaStore.MediaColumns.DateTaken, TimeMillis(dateTimeNow));
+                values.Put(MediaStore.MediaColumns.IsPending, true);
+
+                var uri = context.ContentResolver.Insert(externalContentUri, values);
+                using var stream = context.ContentResolver.OpenOutputStream(uri);
+                fileStream.CopyTo(stream);
+                stream.Close();
+
+                values.Put(MediaStore.MediaColumns.IsPending, false);
+                context.ContentResolver.Update(uri, values, null, null);
+#else
+                throw new Exception("For using save to gallery on android 10 and newer, use the target TargetFrameworks MonoAndroid10.0 or MonoAndroid11.0");
+#endif
+            }
+            else
+            {
+#pragma warning disable CS0618 // Type or member is obsolete
+                using var directory = new File(Environment.GetExternalStoragePublicDirectory(relativePath), albumName);
+
+                directory.Mkdirs();
+                using var file = new File(directory, newFileName);
+
+                using var fileOutputStream = System.IO.File.Create(file.AbsolutePath);
+                fileStream.CopyTo(fileOutputStream);
+                fileOutputStream.Close();
+
+                values.Put(MediaStore.MediaColumns.Data, file.AbsolutePath);
+                context.ContentResolver.Insert(externalContentUri, values);
+
+                var mediaScanIntent = new Intent(Intent.ActionMediaScannerScanFile);
+                mediaScanIntent.SetData(Uri.FromFile(file));
+                context.SendBroadcast(mediaScanIntent);
+#pragma warning restore CS0618 // Type or member is obsolete
+            }
+        }
+
+        static readonly DateTime jan1st1970 = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        // JavaSystem.CurrentTimeMillis()
+
+        static long TimeMillis(DateTime current)
+            => (long)CalcTimeDifference(current).TotalMilliseconds;
+
+        static long TimeSeconds(DateTime current)
+            => (long)CalcTimeDifference(current).TotalMilliseconds;
```

Comment:
Guess this one should be `TotalSeconds`.

Reply:
oh, yes